### PR TITLE
bugfix_LWDEV-7959-bugs-in-input: added big.js method for addition, fi…

### PR DESCRIPTION
**Pull request by Georg3MK**
…xed number over accuracy when paste value, fixed unnecessary dot issue, removed cursor pointer when balance equal 0

### CODE REVIEW

File: src/components/Order/OrderPercentage.tsx [hunk 1]
```diff
@@ -9,78 +8,19 @@ interface OrderPercentageProps {
...
   isActive,
   onClick,
   isDisabled
 }) => {
   return (
-    <StyledPercent
-      disabled={isDisabled}
-      className={isActive ? 'active' : ''}
+    <Percent
+      className={`${isActive ? 'active' : ''} ${isDisabled ? 'disabled' : ''}`}
```
Comment (PaulRumyantsevItr) [inline src/components/Order/OrderPercentage.tsx]: What do you think about using https://github.com/JedWatson/classnames lib?

File: src/components/Order/styles.ts [hunk 2]
```diff
@@ -212,3 +214,86 @@ export const Link = styled.a`
   cursor: pointer;
   text-decoration: underline;
 `;
+
+export const Percent = styled.div`
+  color: #f5f6f7;
```
Comment (PaulRumyantsevItr) [inline src/components/Order/styles.ts]: colors.whiteText

File: src/components/Order/OrderLimit.tsx [hunk 3]
```diff
@@ -90,7 +90,10 @@ const OrderLimit: React.SFC<OrderLimitProps> = ({
       <InputControl>
         <Flex justify={'space-between'} style={{marginBottom: '7px'}}>
           <Action>{`Amount (${baseAssetName})`}</Action>
-          <Available onClick={onHandlePercentageChange()}>
+          <Available
+            onClick={onHandlePercentageChange()}
+            className={!balance ? 'disabled' : ''}
```
Comment (sergeyschegolev) [inline src/components/Order/OrderLimit.tsx]: Pls, use props inside the SC

File: src/stores/uiOrderStore.ts [hunk 4]
```diff
@@ -84,11 +84,11 @@ class UiOrderStore extends BaseStore {
   setPriceValueWithFixed = (price: number) =>
     (this.priceValue = !price
       ? DEFAULT_INPUT_VALUE
-      : price.toFixed(this.priceAccuracy));
-  setQuantityValueWithFixed = (quantity: number) =>
+      : bigToFixed(price, this.priceAccuracy));
+  setQuantityValueWithFixed = (quantity: any) =>
```
Comment (PaulRumyantsevItr) [inline src/stores/uiOrderStore.ts]: Why it's any now? You can list available types like ```number | string```

File: src/components/Order/OrderPercentage.tsx (hunk 1 above)
Comment (sergeyschegolev) [inline src/components/Order/OrderPercentage.tsx]: the same, pls

File: src/utils/__tests__/math.spec.ts [hunk 5]
```diff
@@ -35,16 +36,24 @@ describe('Test math functions', () => {
...
+    const result = '1111111111.00000001';
+
+    expect(addition(term1, term2)).toBe(result);
+  });
+
   it('should subtract decrement from number and return correct value', () => {
-    let value = 0.3;
-    let decrement = 0.1;
-    let result = 0.2;
+    let value = '0.3';
```
Comment (PaulRumyantsevItr) [inline src/utils/__tests__/math.spec.ts]: I don't think it's correct that math lib return string as a result of number calculations

File: src/utils/inputNumber.ts [hunk 6]
```diff
@@ -34,9 +36,12 @@ export const onValueChange = (setValue: any, getAcc: any, value: string) => {
   }
   value = substringZero(value);
   value = substringMinus(value);
+  if (getAcc() === 0) {
```
Comment (PaulRumyantsevItr) [inline src/utils/inputNumber.ts]: What is Acc?

File: src/utils/__tests__/math.spec.ts (hunk 5 above)
Comment (Georg3MK) [reply src/utils/__tests__/math.spec.ts]: We use big.js for some operations in math lib that returns string value. It's safe way to get correct values from calculations in some rare cases like 1111111111.00000008 (number will be rounded to 1111111111 by JS automatically). So I believe we should keep strings here.

File: src/components/Order/OrderPercentage.tsx [hunk 7]
```diff
@@ -1,6 +1,8 @@
-import rem from 'polished/lib/helpers/rem';
 import * as React from 'react';
-import styled from 'styled-components';
+import {Percent} from './styles';
+
+// tslint:disable-next-line:no-var-requires
+const classNames = require('classnames');
```
Comment (PaulRumyantsevItr) [inline src/components/Order/OrderPercentage.tsx]: Add @types/classnames and use import

**Pull request closed without merging**